.github/workflows/template_release_version.yml

```diff
@@ -0,0 +1,46 @@
+name: Release Version Detector
+
+on:
+  workflow_call:
+    outputs:
+      new_version:
+        value: ${{ jobs.new_version.outputs.new_version }}
+      new_tag:
+        value: ${{ jobs.new_version.outputs.new_tag }}
+
+jobs:
+  new_version:
+
+    name: Get new release version
+    runs-on: ubuntu-22.04
+
+    outputs:
+      new_version: ${{ steps.set_version.outputs.next_version }}
+      new_tag: ${{ steps.set_version.outputs.next_tag }}
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3.6.0
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Detect new release version
+        id: set_version
+        run: |
+          # fetch last tag from git
+          OLD_VERSION=`git describe --tags $(git rev-list --tags --max-count=1)`
```

Comment:
We could fetch the old version without checking out the code and the tags:

```shell
gh release list | grep Latest
```

Will return for `alarm`

```bash
1.13.0	Latest	1.13.0	2023-07-12T11:56:37Z
```

So we can elaborate the GitHub API which is accessible within the Workflow to fetch the latest version. This will reduce the time for checking out big repositories.